### PR TITLE
Use URLSearchParams to serialize

### PR DESCRIPTION
**Pull request by razor-x**
- Pass paramsSerializer
- Use URLSearchParams for paramsSerializer

### CODE REVIEW

File: src/lib/params-serializer.test.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+import {
+  paramsSerializer,
+  UnserializableParamError,
+} from './params-serializer.js'
+
+test('serializes empty object', (t) => {
+  t.is(paramsSerializer({}), '')
+})
+
```
Comment (chamokk) [inline src/lib/params-serializer.test.ts:11]: Should we probably add test cases for checking params as null or undefined?

Comment (razor-x) [reply src/lib/params-serializer.test.ts:11]: This function implements the axios paramsSerializer which only takes `Record` type, so if params is null or undefined this function should not get called.

File: src/lib/params-serializer.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import type { CustomParamsSerializer } from 'axios'
+
+export const paramsSerializer: CustomParamsSerializer = (params) => {
+  const searchParams = new URLSearchParams()
+
```
Comment (chamokk) [inline src/lib/params-serializer.ts:5]: Should we check for falsy values (null, undefined) before destructuring in the for loop?
